### DISCUSSION TITLE
Supporting arbitrary values in theme extension?

### DISCUSSION THREAD

**Discussion by tettoffensive** [tailwindlabs/tailwindcss#9433] [Help]
Is there a way for me to support using arbitrary values in a theme extension? For example, rather than hard-coding the '300px'  value below, I'd like to let the user type in `grid-cols-auto-fit-[300px]` or something like that
**tailwind.config.js:**
```js
theme: {
  extend: {
    gridTemplateColumns: {
      'auto-fit': 'repeat(auto-fit, minmax(min(300px, 100%), 1fr))'
    },
  }
}
```

**Comment by wongjn** [comment]
You could write a plugin and use `matchUtilities` with the prefix of `grid-cols-auto-fit`:
```js
module.exports = {
  theme: {
    extend: {
      // ...
    },
  },
  plugins: [
    require('tailwindcss/plugin')(({ matchUtilities }) => {
      matchUtilities({
        'grid-cols-auto-fit': (value) => ({
          'grid-template-columns': `repeat(auto-fit, minmax(min(${value}, 100%), 1fr))`,
        }),
      });
    }),
  ],
}
```
See this [Tailwind Play](https://play.tailwindcss.com/VWe07kHEx1?file=config) of this code working (be sure to open up the Generated CSS panel).

**Reply by tettoffensive** [reply]
Thanks, I was wondering if it involved a plug-in or there was some built-in syntax. I'll try the plug-in route
